### ISSUE TITLE
Mentions: unexpected behavior with max-height and scroll

### ISSUE THREAD

**Issue by lagroms** [ant-design/ant-design#41654]
### Reproduction link

[https://stackblitz.com/edit/react-xhqjdv?file=demo.tsx,index.css](https://stackblitz.com/edit/react-xhqjdv?file=demo.tsx,index.css)

### Steps to reproduce

1. add a max-height on the textarea (already done in the reproduction link)
2. add enough text to enable the scroll in the textarea
3. trigger the suggestion panel by typing "@"
4. scroll up

![antd-bug](https://user-images.githubusercontent.com/88441919/230060336-21a67a67-50ab-4716-8d8a-745abb9a6170.gif)

### What is expected?

- have the suggestion panel to have a "sticky" behavior where it wont move at all when scrolling
OR 
- stay at same place where it has been triggered, but automatically hides when scrolling up (the same way Facebook is using it)

![facebook-scroll](https://user-images.githubusercontent.com/88441919/230060305-32a70928-6f97-4f00-bcad-83357cc1dc27.gif)

### What is actually happening?

stay at same place where it has been triggered but stays visible

| Environment | Info |
| --- | --- |
| antd | 5.4.0 |
| React | 17.0.4 |
| System | macOS 13.3 |
| Browser | Chome 111.0.5563.146 |

**Comment by CooperHash** [comment]
no one working on it？

**Comment by rickvian** [comment]
i don't think you can expect it to stay on the place, what if there are more content below and user scroll down?
another user may expect the container to follow the '@' tag 

to be more precise, i think the ideal behaviour would be follow the '@' but stop when it hits the top edge / bottom edge
